arangod/Aql/HashedCollectExecutor.cpp

```diff
@@ -314,11 +314,29 @@ decltype(HashedCollectExecutor::_allGroups)::iterator HashedCollectExecutor::fin
   }
 
   _nextGroupValues.clear();
-  // for inserting into group we need to clone the values
-  // and take over ownership
-  for (auto const& reg : _infos.getGroupRegisters()) {
-    _nextGroupValues.emplace_back(input.stealValue(reg.second));
+
+  if (_infos.getGroupRegisters().size() == 1) {
+    for (auto const& reg : _infos.getGroupRegisters()) {
+      // On a single register there can be no duplicate value
+      // inside the groupValues, so we cannot get into a situation
+      // where it is unclear who is responsible for the data
+      _nextGroupValues.emplace_back(input.stealValue(reg.second));
```

Comment:
I know the code has been like this before, but in theory this can leak if `stealValue()` works (which it will do) but the `emplace_back()` fails with OOM.
In this case the dynamic memory pointed to by the AqlValue is leaked.
While we are here, we should probably fix this too.